src/Akka.Persistence.Redis/Journal/RedisJournal.cs

```diff
@@ -1,82 +1,120 @@
-﻿using System;
+﻿//-----------------------------------------------------------------------
+// <copyright file="RedisJournal.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
 using System.Threading.Tasks;
 using Akka.Actor;
 using Akka.Persistence.Journal;
-using RedisBoost;
-using RedisBoost.Core.Serialization;
+using Akka.Serialization;
+using Akka.Util.Internal;
+using StackExchange.Redis;
 
 namespace Akka.Persistence.Redis.Journal
 {
     public class RedisJournal : AsyncWriteJournal
     {
         private readonly RedisSettings _settings = RedisPersistence.Get(Context.System).JournalSettings;
-        private readonly IRedisClientsPool _pool;
-        private readonly BasicRedisSerializer _serializer;
-
-        public RedisJournal()
-        {
-            _serializer = new RedisPersistenceSerializer(Context.System);
-            _pool = RedisClient.CreateClientsPool();
-        }
+        private Lazy<Serializer> _serializer;
+        private Lazy<IDatabase> _database;
+        private ActorSystem _system;
 
-        protected override void PostStop()
+        protected override void PreStart()
         {
-            base.PostStop();
-
-            _pool.Dispose();
+            base.PreStart();
+            _system = Context.System;
+            _database = new Lazy<IDatabase>(() =>
```

Comment:
No. I can't fix it. IDatabase doesn't implement IDisposable. ConnectionMultiplexer does, but I use it inside lazy only